.github/workflows/ci.yaml

```diff
@@ -0,0 +1,472 @@
+          java-version: '8'
+          distribution: 'temurin'
+
+
+      - name: Setup and execute Gradle 'check' task
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: check
+
+      - name: Save Gradle test reports
+        if: always()
+        run: |
+          mkdir -p /tmp/gradle_test_reports/core
+          mkdir -p /tmp/gradle_test_reports/server
+          mkdir -p /tmp/gradle_test_reports/schema-loader
+          cp -a core/build/reports/tests/test /tmp/gradle_test_reports/core/
+          cp -a server/build/reports/tests/test /tmp/gradle_test_reports/server/
+          cp -a schema-loader/build/reports/tests/test /tmp/gradle_test_reports/schema-loader/
+
+      - name: Save SpotBugs reports
+        if: always()
+        run: |
+          mkdir -p /tmp/gradle_spotbugs_reports/core
+          mkdir -p /tmp/gradle_spotbugs_reports/server
+          mkdir -p /tmp/gradle_spotbugs_reports/schema-loader
+          cp -a core/build/reports/spotbugs /tmp/gradle_spotbugs_reports/core/
+          cp -a server/build/reports/spotbugs /tmp/gradle_spotbugs_reports/server/
+          cp -a schema-loader/build/reports/spotbugs /tmp/gradle_spotbugs_reports/schema-loader/
+
+      - name: Upload Gradle test reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: gradle_test_reports
+          path: /tmp/gradle_test_reports
+
+      - name: Upload Spotbugs reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: gradle_spotbugs_reports
+          path: /tmp/gradle_spotbugs_reports
+
+  integration-test-for-cassandra:
+    runs-on: ubuntu-latest
+
+    services:
+      cassandra:
+        image: cassandra:3.11.11
+        env:
+          MAX_HEAP_SIZE: 2048m
+          HEAP_NEWSIZE: 512m
+        ports:
+          - 9042:9042
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Setup and execute Gradle 'integrationTestCassandra' task
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: integrationTestCassandra
+
+      - name: Save Gradle test reports
+        if: always()
+        run: |
+          mkdir -p /tmp/gradle_integration_test_reports/core
+          mkdir -p /tmp/gradle_integration_test_reports/schema-loader
+          cp -a core/build/reports/tests/integrationTestCassandra /tmp/gradle_integration_test_reports/core/
+          cp -a schema-loader/build/reports/tests/integrationTestCassandra /tmp/gradle_integration_test_reports/schema-loader/
+
+      - name: Upload Gradle test reports
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: gradle_cassandra_integration_test_reports
+          path: /tmp/gradle_integration_test_reports
+
+  integration-test-for-cosmos:
+    runs-on: self-hosted
```

Comment:
We decided to limit the number of self-hosted runners to two. With 3 runners running the test in parallel, there are a lot of cosmos DB request metadata failure making the whole test fail.